python/neutron-understack/neutron_understack/nautobot.py

```diff
@@ -86,6 +87,20 @@ def ucvni_delete(self, network_id):
         url = f"/api/plugins/undercloud-vni/ucvnis/{network_id}/"
         return self.make_api_request(url, "delete")
 
+    def fetch_namespace_by_name(self, name: str) -> str:
+        url = f"/api/ipam/namespaces/?name={name}&depth=1"
+        resp_data = self.make_api_request(url, "get")
+        return resp_data["results"][0]["id"]
```

Comment:
What if it's not found? Does this return an empty list which means we throw an IndexError exception here?

Reply:
fixed, we will raise Nautobot error